cachix/src/Cachix/Client/InstallationMode.hs

```diff
@@ -122,18 +147,19 @@ addBinaryCache config bc _ (Install ncl) = do
       NixConf.Custom _ -> do
         lnc <- NixConf.read ncl
         return ([lnc], lnc)
-  let nixconf = fromMaybe (NixConf.NixConf []) output
-  netrcLocMaybe <- forM (guard $ not (BinaryCache.isPublic bc)) $ const $ addPrivateBinaryCacheNetRC config bc ncl
-  let addNetRCLine :: NixConf.NixConf -> NixConf.NixConf
-      addNetRCLine = fromMaybe identity $ do
-        netrcLoc <- netrcLocMaybe :: Maybe FilePath
-        -- We only add the netrc line for local user configs for now.
-        -- On NixOS we assume it will be picked up from the default location.
-        guard (ncl == NixConf.Local)
-        pure (NixConf.setNetRC $ toS netrcLoc)
-  NixConf.write ncl $ addNetRCLine $ NixConf.add bc (catMaybes input) nixconf
+  return (catMaybes input, fromMaybe (NixConf.NixConf []) output)
+
+removeBinaryCache :: Text -> InstallationMode -> IO ()
+removeBinaryCache name (Install ncl) = do
+  (input, output) <- prepareNixConf ncl
+  let (final, removed) = NixConf.remove name input output
+  NixConf.write ncl final
   filename <- NixConf.getFilename ncl
-  putStrLn $ "Configured " <> BinaryCache.uri bc <> " binary cache in " <> toS filename
+  if removed
+    then putStrLn $ "Removed https://" <> name <> ".cachix.org binary cache in " <> toS filename
```

Comment:
we should probably get the full uri from the config that was removed and show it here